src/js/WinJS/Accents.ts

```diff
@@ -0,0 +1,171 @@
+                // inverseBody = "color: *accent*; background-color: *listSelectHoverInverse"
+
+                selectorSplit.forEach(sel => {
+                    css += "\n" + inverseThemeSelector + " " + sel;
+                    if (CSSSelectorTokens.indexOf(sel[0]) !== -1) {
+                        css += ",\n" + inverseThemeSelector + sel;
+                    }
+                    css += " {\n" + inverseBody + "\n}";
+                });
+                // css
+                //.foo, .bar:hover, div:hover, html.win-hoverable .bar:hover, html.win-hoverable.bar:hover {body} 
+                //.win-ui-light .foo {inverseBody}
+                //.win-ui-light.foo {inverseBody}
+                //.win-ui-light .bar:hover {inverseBody}
+                //.win-ui-light.bar:hover {inverseBody}
+                //.win-ui-light div:hover {inverseBody}
+            }
+            return css;
+        }).join("\n");
+        _Global.document.head.insertBefore(style, _Global.document.head.firstChild);
+    });
+}
+
+function handleColorsChanged() {
+    var UIColorType = _WinRT.Windows.UI.ViewManagement.UIColorType;
+    var accent = colorToString(UISettings.getColorValue(_WinRT.Windows.UI.ViewManagement.UIColorType.accent), 1);
+    if (colors[0] === accent) {
+        return;
+    }
+
+    // Establish colors
+    colors = [];
+    colors.push(accent);
+    colors.push(
+        colorToString(UISettings.getColorValue(_WinRT.Windows.UI.ViewManagement.UIColorType.accent), (isDarkTheme ? 0.6 : 0.4)),
+        colorToString(UISettings.getColorValue(_WinRT.Windows.UI.ViewManagement.UIColorType.accent), (isDarkTheme ? 0.8 : 0.6)),
+        colorToString(UISettings.getColorValue(_WinRT.Windows.UI.ViewManagement.UIColorType.accent), (isDarkTheme ? 0.9 : 0.7)),
+        colorToString(UISettings.getColorValue(_WinRT.Windows.UI.ViewManagement.UIColorType.accent), (isDarkTheme ? 0.4 : 0.6)),
+        colorToString(UISettings.getColorValue(_WinRT.Windows.UI.ViewManagement.UIColorType.accent), (isDarkTheme ? 0.6 : 0.8)),
+        colorToString(UISettings.getColorValue(_WinRT.Windows.UI.ViewManagement.UIColorType.accent), (isDarkTheme ? 0.7 : 0.9)));
+
+    scheduleWriteRules();
+}
+
+function colorToString(color: _WinRT.Windows.UI.Color, alpha: number) {
+    return "rgba(" + color.r + "," + color.g + "," + color.b + "," + alpha + ")";
+}
+
+function cleanup() {
+    var style = _Global.document.head.querySelector("#" + Constants.accentStyleId);
+    style && style.parentNode.removeChild(style);
+}
+
+function _reset() {
+    rules = [];
+    cleanup();
+}
+
+_BaseUtils.ready().then(() => {
+    // Figure out color theme
+    var tag = _Global.document.createElement("winjs");
+    _Global.document.body.appendChild(tag);
+    var theme = _Global.getComputedStyle(tag).opacity;
+    isDarkTheme = theme === "0";
+    tag.parentElement.removeChild(tag);
+
+    if (_BaseUtils.hasWinRT) {
```

Comment:
Should feature detect like this: `if (_WinRT.Windows.UI.ViewManagement.UISettings)`